### PR TITLE
[CVSRP-4580] update and refactor Regular Pay Reference Pay to be one …

### PR DESCRIPTION
**Pull request by peckoverHMRC**
…dynamic page that handles all employee types

### CODE REVIEW

**Comment by peckoverHMRC** [comment]
Type1

Type 1

**Comment by peckoverHMRC** [comment]
Type2a

Type 2a

**Comment by peckoverHMRC** [comment]
Type2bWelsh

Type 2b Welsh

File: conf/messages.cy [hunk 1]
```diff
@@ -251,12 +251,9 @@ reviewPayDates.heading = Dyddiadau cyflog o fewn y cyfnod hawlio
...
 
-regularPayAmount.title = What was the employee paid in the last pay period ending on or before 19 March 2020?
-regularPayAmount.title.ext = What was the employee paid in the last pay period ending on or before 30 October 2020?
-regularPayAmount.heading = What was the employee paid in the last pay period ending on or before 19 March 2020?
-regularPayAmount.heading.ext = What was the employee paid in the last pay period ending on or before 30 October 2020?
-regularPayAmount.p1 = This is the gross amount paid in the last pay period ending on or before 19 March, before deductions.
-regularPayAmount.p1.ext = This is the gross amount paid in the last pay period ending on or before 30 October, before deductions.
+regularPayAmount.title = What was the employee paid in the last pay period ending on or before {0}?
+regularPayAmount.heading = What was the employee paid in the last pay period ending on or before {0}?
+regularPayAmount.p1 = This is the gross amount paid in the last pay period ending on or before {0}, before deductions.
```
Comment (LewisMitchell) [inline conf/messages.cy:256]: I think there is a missing message `regularPayAmount.label`?

File: it/controllers/RegularPayAmountControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      .value
+
+    getTests(mar2nd2021, userAnswers)
+  }
+
+  "POST /regular-pay-amount" when {
+
+    "enters a valid answer" when {
+
+      "redirect to LastPayDate page" in {
```
Comment (LewisMitchell) [inline it/controllers/RegularPayAmountControllerISpec.scala:58]: I think this needs to be changed to 'PartTimeQuestion page'

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Acceptance tests fail for me 😢

**Comment by peckoverHMRC** [comment]
ATs pass against master

**Review by LewisMitchell — APPROVED** [review]
LGTM. Just running AT's.

**Pull request merged**